[Tizen.Applications.AttachPanel] Add attach_panel feature checking routine

hallower opened the pull request:
### Description of Change ###

        . Add attach_panel feature checking routine for Tizen.Applications.AttachPanel APIs.
        . Add description for missing exceptions.
        . Rearragement API implementation

    b) Person in charge
         ChulSeung Kim (charles0.kim@samsung.com)
         Young-Hun Jo (hoon.cho@samsung.com)
         JinYoung Kim (jin0.kim@samsung.com)
         Hyerim Kim (rimi.kim@samsung.com)
         Geunsun Lee (gs86.lee@samsung.com )

### Bugs Fixed ###

N/A

### API Changes ###

If you have the ACR for changing APIs, put the link of the ACR:
 - ACR: Not yet

If you don't have the ACR, List all API changes here (or just put None), example:

Added:
  d) Feature
        http://tizen.org/feature/attach_panel
  
    e) Complete ACR contents
	CS files containing the changes
        	- AttachPanel.cs
		- AttachPanelInternal.cs

	List of API changes

	[Constructors]
	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
        	public AttachPanel(EvasObject conformant)

	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
        	public AttachPanel(Conformant conformant)


	[Properties]
	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
        	public StateType State

	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
        	public bool Visible


	[Methods]
	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
	          public void AddCategory(ContentCategory category, Bundle extraData)

	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
	          public void RemoveCategory(ContentCategory category)
	
	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
	          public void SetExtraData(ContentCategory category, Bundle extraData)

	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
	          public void Show()

	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
	          public void Show(bool animation)

	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
	          public void Hide()

	+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
	          public void Hide(bool animation)


### Behavioral Changes ###

Adding this change, user can get explicit NotSupportedException when the user would like to use AttachPanel APIs on the unsupported devices or profiles.

myroot commented on src/Tizen.Applications.AttachPanel/Tizen.Applications.AttachPanel/AttachPanel.cs:
```diff
@@ -79,12 +101,10 @@ public AttachPanel(Conformant conformant)
-                Interop.AttachPanel.ErrorCode err = Interop.AttachPanel.DestroyAttachPanel(_attachPanel);
-                CheckException(err);
-                _attachPanel = IntPtr.Zero;
+                Interop.AttachPanel.DestroyAttachPanel(s_attachPanel);
```
It internally call EFL API. So, It can be issue run on GC thread. 
(EFL is not thread safe)

hallower replied:
I have to check this out later with registrating as a JIRA issue.
Thanks

myroot commented on src/Tizen.Applications.AttachPanel/Tizen.Applications.AttachPanel/AttachPanel.cs:
```diff
@@ -13,29 +13,40 @@ public partial class AttachPanel
+            }
+
+            var candidateAttachPanel = new IntPtr();
+            var err = Interop.AttachPanel.CreateAttachPanel(conformant, ref candidateAttachPanel);
```
How abut use `out` instead of `ref`

hallower replied:
done

myroot commented on src/Tizen.Applications.AttachPanel/Tizen.Applications.AttachPanel/AttachPanel.cs:
```diff
@@ -13,29 +13,40 @@ public partial class AttachPanel
+                CheckException(Interop.AttachPanel.ErrorCode.AlreadyExists);
+            }
+
+            var candidateAttachPanel = new IntPtr();
```
IntPtr.Zero is more explicit instead of `new IntPtr()` because IntPtr is a value type

hallower replied:
done

myroot commented on src/Tizen.Applications.AttachPanel/Tizen.Applications.AttachPanel/AttachPanel.cs:
```diff
@@ -48,27 +59,38 @@ public AttachPanel(EvasObject conformant)
         /// <exception cref="InvalidOperationException">Thrown when the AttachPanel already exists or the <paramref name="conformant"/> is not a conformant object.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the AttachPanel is not supported in the device.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when the parameter is null</exception>
         public AttachPanel(Conformant conformant)
```
It is redundancy code, because Conformant is EvasObject

```c#
        public AttachPanel(Conformant conformant) : this(conformant as EvasObject)
        {
        }
```
It is enough

hallower replied:
Yes totally I agree with that. done.

hallower commented:
Thank you for reviewing, I will merge this pull request during ACR process.
Also, I will make a new pull request regarding Extra things

hallower commented:
I added feature description to APIs and changed some of reviewed issues.
Regarding static event handler and instnace destroying issue, I will fix later with another pull request.
Thank you.
